src/clj/puppetlabs/services/jruby/jruby_internal.clj

```diff
@@ -139,39 +146,42 @@
 (schema/defn ^:always-validate
   cleanup-pool-instance!
   "Cleans up and cleanly terminates a JRubyInstance and removes it from the pool."
-  [{:keys [scripting-container pool] :as instance} :- JRubyInstance]
+  [{:keys [scripting-container pool] :as instance} :- jruby-schemas/JRubyInstanceSchema
+   shutdown-fn :- IFn]
   (.unregister pool instance)
-  ;; TODO: need to add support for a callback hook, so that consumers like
-  ;; puppet-server can do their own cleanup.
-  ;(.terminate jruby-puppet)
+  (shutdown-fn instance)
```

Comment:
I'm assuming you mean `cleanup-pool-instance!`... yes, probably, same as `create-pool-instance!` I've been holding off on moving stuff around because I think there's some other stuff in the same vein we'll want to do for removing the TK service and adding a pool manage service, and since I feel like there's still a little uncertainty there as to where things will end up, I was thinking of holding off on any reorganization until we start work on that ticket.